### PR TITLE
improves install process using setuptools and pyproject.toml

### PR DESCRIPTION
**Pull request by kevindamm-jigsaw**
More explicit declaration of dependencies.  Allows pip install and adds `rids` CLI command to path, and makes it easier for tests to refer to rids package despite being in a sibling or cousin directory.

Follow-up commit will include moving the source files around and modifying the install.sh to work with `pip install ...rids`

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+dependencies = [
+  "absl-py",
+]
+
+[build-system]
+requires = ["setuptools", "wheel"]
+build-backend = "setuptools.build_meta"
+
+[tools.setuptools]
+packages = ["pyrids"]
```
Comment (fortuna) [inline pyproject.toml:16]: Don't you need to define that package in the source code?

**Review by fortuna — APPROVED** [review]

File: pyproject.toml (hunk 1 above)
Comment (kevindamm-jigsaw) [reply pyproject.toml:16]: yes, that's part of the follow-up commit mentioned in the PR description -- it requires moving around files and a significant change to how install.sh works and I'd like to get the service calling `rids` directly too now that we don't need the shell script for any setup.  I figured I'd give you a chance to comment on the .toml file first, though.

**Comment by fortuna** [comment]
I think the prefix py on pyrids is redundant. Let's drop it.

**Comment by kevindamm-jigsaw** [comment]
Yeah, I was on the fence about it -- since we've already talked about translating it into Go, I wanted to avoid any issues/confusion with source directory names and how we refer to each.. but since PyPI won't ever know about the golang version and Go won't ever know about the python version, it's probably a non-issue.

I'll rename pyrids to rids.  I'm building and uploading to testpip to verify that build/package/install works as expected.  I'll update this PR after I've checked everything out.

**Review by fortuna — APPROVED** [review]
Nice!

**Pull request merged**